File: java/src/FirstNonConsecutive.java
public class FirstNonConsecutive {
    static Integer find(final int[] array) {

        int first = array[0];

        for (int i = 1; i < array.length; i++) {
            if (array[i] != array[i - 1] + 1) {
                first = array[i];
                break;
            }

        }

        if (first == array[0]) {
            return null;
        }

        return first;
    }

    public static void main(String[] args) {
        int[] array = {1, 2, 3, 4, 6, 7, 8,};
        System.out.println(find(array));
    }
}

File: javascript/first-non-consecutive.js
function firstNonConsecutive (arr) {
  let first = arr[0];
  
  for (let i = 1; i < arr.length; i++) {
    if (arr[i] != arr[i-1] + 1) {
      first = arr[i];
      break;
    }
  }
  
  if (first == arr[0]) {
    return null;
  }
  
  return first;
}

console.log(firstNonConsecutive([1, 2, 3, 4, 6, 7, 8]));